src/tasks

```diff
@@ -1,32 +1,32 @@
 1. Which of the following SOLID principles are violated here?
 (Multiple choice)
-* Single Responsibility Principle (yes)
-* Open-Close principle (yes)
-* Liskov Substitution Principle (no)
-* Inversion of Control (no, tricky, since it's not a SOLID principle :) )
-* Dependency Inversion Principle (yes) 
+* Single Responsibility Principle (yes) [There are multiple reason to modify: print, order, composition]
+* Open-Close principle (yes) [PrintingService is hard-wired, we can't extend it without modifying the NameService]
+* Liskov Substitution Principle (no) [No inheritance]
+* Inversion of Control (no) [Tricky, since it's not a SOLID principle :)]
```

Comment:
but otherwise violated?

Reply:
Yes. The PrintingService gets the control in procedural way.